### ISSUE TITLE
CKEditor: Events not fired anymore after changing the mode

### ISSUE THREAD

**Issue by blutorange** [primefaces-extensions/primefaces-extensions.github.com#732]
To see this, change the `Basic Usage` showcase for the CKEditor like this by adding an AJAX listener for the `change` event:

![image](https://user-images.githubusercontent.com/2456413/65982363-782f8000-e47b-11e9-83d6-5d69309c58c0.png)

```xml
<p:growl id="growl" showDetail="true" />  
  
<pe:ckEditor id="editor" value="#{editorController.content}" interfaceColor="#{editorController.color}">  
    <p:ajax event="save" listener="#{editorController.saveListener}" update="growl"/>  
    <p:ajax event="change" onstart="console.log('change')"/>  
</pe:ckEditor>  
  
<p:commandButton actionListener="#{editorController.changeColor}" update="@form:centerExample"  
                 value="Change color with AJAX" style="margin-top:10px;"/>  
```

When you type anything in the CKEditor, the message `change` is logged. But if you, for example, switch to the source code mode and then switch back, it won't work anymore.

---

The reason for this lies in the way the event listeners are attached to the CKEditor in the widget. It's explained [here on the CKEditor docs](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_editable.html#method-attachListener) and also [here on stackexchange](https://stackoverflow.com/a/22443086).

Events are currently attached like this:

```javascript
        this.instance.editable().attachListener(editable, 'paste', $.proxy(function(event) {
                this.checkChange();
                this.checkDirty();
        }, this));
```

 The `editable` refers to the DOM element of the editor content. This element is replaced, for exapmle, when you change the mode of the CKEditor. As mentioned in the docs, this should probably be changed to

```javascript
        //changes to WYSIWYG mode
    	this.instance.on('contentDom', $.proxy(function(event) {
            this.bindEditorEvents(this.instance);
            this.fireEvent('wysiwygMode');
      	}, this));
```

with the `bindEditorEvents` method containing the parts where the events listeners are attached to the `editor.editable()` instance:

Also, `contentDom` is fired only when the editor is switch to WYSIWYG mode. To make the events work in source code mode, [they can be bound like this](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_editor.html#event-change):

```javascript
    	//changes to source mode
    	this.instance.on('mode', $.proxy(function(event) {
                if (this.instance.mode != 'source') {
                    return;
                }
                this.bindEditorEvents(this.instance);
                this.fireEvent('sourceMode');
        }, this));
```

---

I already tried this locally, if you want I can make a PR for this.

**Comment by melloware** [comment]
PR would be awesome.  I will merge it.

**Comment by melloware** [comment]
When you submit PR do you mind adding a change event to the showcase so we have this test in the future?

**Comment by blutorange** [comment]
Probably today when I get home. Sure, I can get add something to the showcase.

**Comment by blutorange** [comment]
Another subtle thing I noticed, there is a `dirty` event that is fired on change, but only when the [CKEditor instance is dirty](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_editor.html#method-checkDirty).

When you try it out, you'll notice the `dirty` event is not fired when you press a key to enter the first character, but only once you enter the second character.

That's because the widget is listening to the `keydown` event. Immediately after a keydown, the CKEditor content has not changed yet, so `checkDirty` report `false`.

Changing `keydown` to `keyup` [on line 279](https://github.com/primefaces-extensions/core/blob/master/src/main/resources/META-INF/resources/primefaces-extensions/ckeditor/widget.js#L279) seems to help, so I'll change that as well.

**Comment by melloware** [comment]
Agreed!  Thanks in advance.

**Comment by melloware** [comment]
You can submit one PR but I created a separate ticket for it: https://github.com/primefaces-extensions/primefaces-extensions.github.com/issues/734
